flyspell-correct.el

```diff
@@ -111,17 +111,26 @@ highlighting."
 ;;; Default interface using `completing-read'
 ;;
 
-(defun flyspell-correct--action (name key label)
-  "Format action string.
-NAME is the action name.
-KEY is the shortcut key.
-LABEL is the action label."
-  (setq key (char-to-string key))
-  (cons (concat (propertize (format "@%s " key) 'invisible t)
-                (replace-regexp-in-string key
-                                          (propertize key 'face '(bold minibuffer-prompt))
-                                          (propertize (format "[%s]" label) 'face 'minibuffer-prompt)))
-        name))
+(defvar flyspell-correct--cr-key "@"
+  "Shortcut key used by `flyspell-correct-completing-read'.")
+
+(defvar flyspell-correct--cr-actions
+  '((save ?s "[Save]")
+    (session ?a "[Accept (session)]")
+    (buffer ?b "[Accept (buffer)]")
+    (skip ?k "[Skip]")
+    (stop ?p "[Stop]"))
+  "Actions used by `flyspell-correct-completing-read'.")
+
+(defun flyspell-correct--cr-index (n)
+  "Generate a short unique index string for N."
+  (let ((str (char-to-string (+ ?1 (mod n 5)))))
```

Comment:
Why `5`? I would rather use length of `flyspell-correct--cr-actions` - more future proof.

Reply:
The `flyspell-correct--cr-index` is used for the indexes prepended before the suggestions. Base 5 is used here to result in nice index strings. The indices are generated such that they terminate with 1,2,3,4,5 such that we can commit if one of these keys are pressed.

~~~ elisp
(mapcar #'flyspell-correct--cr-index (number-sequence 0 100))

("1" "2" "3" "4" "5" "61" "62" "63" "64" "65" "71" "72" "73" "74" "75" "81" "82" "83" "84" "85" "91" "92" "93" "94" "95" "01" "02" "03" "04" "05" "661" "662" "663" "664" "665" "671" "672" "673" "674" "675" "681" "682" "683" "684" "685" "691" "692" "693" "694" "695" "601" "602" "603" "604" "605" "761" "762" "763" "764" "765" "771" "772" "773" "774" "775" "781" "782" "783" "784" "785" "791" "792" "793" "794" "795" "701" "702" "703" "704" "705" "861" "862" "863" "864" "865" "871" "872" "873" "874" "875" "881" "882" "883" "884" "885" "891" "892" "893" "894" "895" "801")
~~~

Reply:
Ah, I see. Misunderstood the intent here. Thanks for explanation.